src/shared-components/button/components/linkButton/index.tsx

```diff
@@ -11,9 +11,9 @@ import type { ButtonType } from '../../types';
 
 export interface LinkButtonProps {
   /**
-   * Specifies the tag or element to be rendered
+   * Specifies the tag or element to be rendered, like an 'a' or 'span'
    */
-  as?: 'a' | React.ElementType;
+  as?: string | React.ElementType;
```

Comment:
The string type literal was being overwritten, so the type was `React.ElementType | undefined`. We also use `span` so the literal definition was incomplete, anyway. This definition is not super strict but should be fine for now.